models/global-registry.js

```diff
@@ -0,0 +1,168 @@
+      key_username: profile.login,
+      first_name: profile.firstName,
+      last_name: profile.lastName,
+      email_address: {
+        client_integration_id: profile.theKeyGuid,
+        email: profile.login
+      },
+      authentication: {
+        client_integration_id: profile.theKeyGuid,
+        key_guid: profile.theKeyGuid
+      },
+      ...(profile.usEmployeeId && !this.isProbablyTestAccount(profile.login) ? {
+        account_number: profile.usEmployeeId,
+        linked_identities: {
+          [PSHR_SYSTEM]: { account_number: profile.usEmployeeId },
+          [SIEBEL_SYSTEM]: { account_number: profile.usEmployeeId }
+        }
+      } : {
+        account_number: null
+      }),
+      ...(designationEntityId ? {
+        'designation:relationship': {
+          designation: designationEntityId,
+          client_integration_id: profile.theKeyGuid
+        }
+      } : {})
+    }
+  }
+
+  /**
+   * If the user's designation has changed since last time, we need to delete the old relationship.
+   * Also, if the relationship is associated with the wrong Key person, we delete it as well.
+   * (Otherwise, we'll get a client_integration_id collision later on.)
+   * Mainly this will happen if the previous step deleted The Key's person entity,
+   * in which case the given user's GR person id must be null.
+   * In Stage, this can also happen if a previous sync deleted the Key person but not the relationship.
+   * (A faulty sync was deployed & run that didn't perform this relationship cleanup step.)
+   * In this case, the given user's GR person id is not null, but it does not match the relationship's person id.
+   */
+  async deleteDesignationRelationshipIfNecessary (profile, deactivated = false) {
+    const relationshipEntity = await this.getDesignationRelationshipEntity(profile.theKeyGuid)
+    if (relationshipEntity) {
+      const grDesignationNumber = get(relationshipEntity, 'designation.designation_number')
+      const currentAssociatedPersonId = get(relationshipEntity, 'person.id')
+      const theKeyPersonId = profile.thekeyGrPersonId
+
+      const shouldDelete =
+        // the user is deactivated
+        deactivated ||
+        // the designation number changed
+        !equalsIgnoreCase(grDesignationNumber, profile.usDesignationNumber) ||
+        // the person entity was recently deleted or doesn't exist yet
+        !theKeyPersonId ||
+        // the person entity doesn't match
+        !equalsIgnoreCase(theKeyPersonId, currentAssociatedPersonId)
+
+      if (shouldDelete) {
+        await this.deleteRelationshipEntity(relationshipEntity.id)
+      }
+    }
+  }
+
+  async getDesignationRelationshipEntity (theKeyGuid) {
+    const result = await this.client.Entity.get({
+      entity_type: PERSON_DESIGNATION_ENTITY_TYPE,
+      filters: { owned_by: THE_KEY_SYSYEM, client_integration_id: theKeyGuid }
+    })
+    /* istanbul ignore else */
+    if (result.entities) {
```

Comment:
can `result.entities` ever be falsy?

Reply:
only on 500 ISE which wouldn't make it here. GR always has an empty entities along with meta if nothing was found.